### PR TITLE
feat(Progress): add single line progress handling and example

### PR DESCRIPTION
**Pull request by kmcfaul**
**What**: If title is not passed in or set to null, the grid gap should not be applied.

**Issue**: #946

### CODE REVIEW

File: packages/patternfly-4/react-core/src/components/Progress/Progress.js [hunk 1]
```diff
@@ -82,6 +82,7 @@ class Progress extends Component {
           getModifier(styles, variant, ''),
           getModifier(styles, measureLocation, ''),
           getModifier(styles, measureLocation === ProgressMeasureLocation.inside ? ProgressSize.lg : size, ''),
+          !title && getModifier(styles, 'singleline', ''),
```
Comment (jschuler) [inline packages/patternfly-4/react-core/src/components/Progress/Progress.js]: this should also work when title defaults to empty string

Comment (kmcfaul) [reply packages/patternfly-4/react-core/src/components/Progress/Progress.js]: Will update. I changed the default of title to null, though a user could still pass an empty string.

Comment (jschuler) [reply packages/patternfly-4/react-core/src/components/Progress/Progress.js]: What I meant was I think you can leave the default to empty string, because in js `!'' === !null`. So shouldn't have to add additional null checks

Comment (kmcfaul) [reply packages/patternfly-4/react-core/src/components/Progress/Progress.js]: Oh I see, good catch. I'll update it

File: packages/patternfly-4/react-core/src/components/Progress/Progress.js [hunk 2]
```diff
@@ -93,7 +94,7 @@ class Progress extends Component {
         <ProgressContainer
           parentId={this.id}
           value={scaledValue}
-          title={title}
+          title={title || ''}
```
Comment (jschuler) [inline packages/patternfly-4/react-core/src/components/Progress/Progress.js]: Can also revert this change since title defaults to empty string
